portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -1,143 +1,155 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package com.google.sps.servlets;
 
-import java.io.IOException;
-import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.users.User; 
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 import com.google.appengine.api.datastore.DatastoreService;
-import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
-import com.google.appengine.api.datastore.FetchOptions; 
-import com.google.appengine.api.datastore.FetchOptions.Builder; 
-import com.google.gson.Gson;
+import com.google.appengine.api.datastore.FetchOptions;
 import com.google.sps.data.Quote; 
+import com.google.sps.data.Util; 
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
 import java.util.List; 
-import java.util.ArrayList; 
+import java.util.ArrayList;
+import java.util.Map; 
+import java.util.HashMap; 
+import java.util.Optional;
 
-/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private static final String QUOTE = "Quote";
-  private static final String TEXT = "text";
-  private static final String TIMESTAMP = "timestamp"; 
+  private static final String INDEX_URL = "/index.html"; 
+
+  // Constants for property of a Quote item in Datastore.
+  private static final String PROPERTY_NAME_QUOTE = "Quote";
+  private static final String PROPERTY_NAME_TEXT = "text";
+  private static final String PROPERTY_NAME_TIMESTAMP = "timestamp"; 
+  private static final String PROPERTY_NAME_USER_EMAIL = "userEmail";
+  private static final String PROPERTY_NAME_NICKNAME = "nickname";
+
+  // Constants for information to be put into response of a GET request.
+  private static final String IS_USER_LOGGED_IN = "loggedIn"; 
```

Comment:
Consider renaming this to,

IS_USER_LOGGED_IN_PARAM

Reply:
Updated.